src/components/molecules/TodoItem.vue

```diff
@@ -0,0 +1,34 @@
+<script setup>
+import AppCheckboxInput from '@/components/atoms/AppCheckboxInput.vue'
+import AppButton from '@/components/atoms/AppButton.vue'
+import { ref } from 'vue'
+
+const props = defineProps({
+  name: { type: String, required: true },
+  completed: { type: Boolean, default: false },
+})
+
+const isHovering = ref(false)
+
+const textDecoration = (isCompleted) => (isCompleted ? { textDecoration: 'line-through', color: '#808080' } : {})
```

Comment:
You can use :hover selector here. I think you don't need this calculation and also these handlers @mouseover="isHovering = true" @mouseout="isHovering = false"

https://developer.mozilla.org/en-US/docs/Web/CSS/:hover